org.scala-refactoring.library/src/main/scala/scala/tools/refactoring/analysis/CompilationUnitDependencies.scala

```diff
@@ -128,6 +144,21 @@ trait CompilationUnitDependencies extends CompilerApiExtensions {
     }
 
     val traverser = new TraverserWithFakedTrees {
+      var annotationTree: Option[Tree] = None
```

Comment:
Me neither, but I didn't find another, easy way to do this because the traverser API is quite imperative (`traverse` returns `Unit` and does not have an accumulator). In my opinion we should have something like `Tree.fold[A](z: A)((acc: A, t: Tree) => A)`. Then we could do tree transversals the functional way. Indeed it should not be hard to implement this on top of the traverser API....